src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt

```diff
@@ -2351,86 +2343,72 @@ namespace Akka.Dispatch
-        protected TSys SystemMessages { get; }
-        protected TUser UserMessages { get; }
-        public override void BecomeClosed() { }
-        public override void CleanUp() { }
-        protected abstract TSys CreateSystemMessagesQueue();
-        protected abstract TUser CreateUserMessagesQueue();
-        protected override void Dispose(bool disposing) { }
-        protected override int GetNumberOfMessages() { }
-        public override void Post(Akka.Actor.IActorRef receiver, Akka.Actor.Envelope envelope) { }
-        protected override void Schedule() { }
+        protected void SetStatus(int newStatus) { }
+        protected bool UpdateStatus(int oldStatus, int newStatus) { }
     }
     public class Mailboxes
     {
         public static readonly string DefaultMailboxId;
         public static readonly string NoMailboxRequirement;
         public Mailboxes(Akka.Actor.ActorSystem system) { }
         public Akka.Actor.DeadLetterMailbox DeadLetterMailbox { get; }
-        public Akka.Dispatch.Mailbox CreateMailbox(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
         public System.Type FromConfig(string path) { }
-        public System.Type GetMailboxType(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
-        public System.Type LookupByQueueType(System.Type queueType) { }
+        public Akka.Dispatch.MailboxType GetMailboxType(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
+        public System.Type GetRequiredType(System.Type actorType) { }
+        public bool HasRequiredType(System.Type actorType) { }
+        public Akka.Dispatch.MailboxType Lookup(string id) { }
+        public Akka.Dispatch.MailboxType LookupByQueueType(System.Type queueType) { }
+        public bool ProducesMessageQueue(System.Type mailboxType) { }
     }
-    [System.FlagsAttribute()]
-    public enum MailboxSuspendStatus
+    public abstract class MailboxType
     {
-        NotSuspended = 0,
-        Supervision = 1,
-        AwaitingTask = 2,
+        protected readonly Akka.Configuration.Config Config;
+        protected readonly Akka.Actor.Settings Settings;
+        protected MailboxType(Akka.Actor.Settings settings, Akka.Configuration.Config config) { }
+        public abstract Akka.Dispatch.MessageQueues.IMessageQueue Create(Akka.Actor.IActorRef owner, Akka.Actor.ActorSystem system);
     }
     public abstract class MessageDispatcher
     {
         public const int DefaultThroughput = 100;
         protected MessageDispatcher(Akka.Dispatch.MessageDispatcherConfigurator configurator) { }
         public Akka.Dispatch.MessageDispatcherConfigurator Configurator { get; }
-        public string Id { get; set; }
+        public static bool DebugDispatcher { get; }
+        public Akka.Event.EventStream EventStream { get; }
+        public string Id { get; }
+        protected long Inhabitants { get; }
```

Comment:
API modifications to `MessageDispatcher`. 

`Inhabitants` is essentially the reference count for each `MessageDispatcher`.